api/src/main/java/info/rmapproject/api/auth/ApiUserServiceImpl.java

```diff
@@ -110,36 +113,64 @@ public URI getSystemAgentUri(String key, String secret) throws RMapApiException
 		try {
 			User user = rmapAuthService.getUserByKeySecret(key, secret);
 			String agentUri = user.getRmapAgentUri();
-			if (agentUri==null){
-				//no agent id
-				throw new RMapApiException(ErrorCode.ER_USER_HAS_NO_AGENT);				
+		    log.debug("Retrieved System Agent as " + ((agentUri==null) ? "" : agentUri));
+			if (agentUri==null || agentUri.length()==0){
+				return null;
 			}
-
-			rmapAuthService.createOrUpdateAgentFromUser(user.getUserId());	
-
 			sysAgentUri = new URI(agentUri);
-			if (!rmapService.isAgentId(sysAgentUri)){
-				//there is no agent id and no flag to create one
-				throw new RMapApiException(ErrorCode.ER_USER_HAS_NO_AGENT);
-			}
+			
 
-		} catch (RMapException e) {
-			e.printStackTrace();
-		} catch (RMapDefectiveArgumentException e) {
-			e.printStackTrace();
 		} catch (RMapAuthException ex) {
 			throw RMapApiException.wrap(ex, ErrorCode.ER_USER_AGENT_COULD_NOT_BE_RETRIEVED);
 		}  catch (URISyntaxException ex) {
 			throw RMapApiException.wrap(ex, ErrorCode.ER_INVALID_AGENTID_FOR_USER);
-		} finally {
-			rmapService.closeConnection();			
```

Comment:
I made some notes about connection management in #123, so I don't think I need to discuss here!